kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java

```diff
@@ -446,6 +447,162 @@ public void buildIpv6SettingsShouldThrowIllegalArgumentExceptionWhenGivenExpecte
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild8021xSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenIllegalArgumentExceptionThrown();
+    }
+
+    @Test
+    public void build8021xSettingsTtlsAndMschapV2() {
+        givenMapWith("net.interface.wlan0.config.802-1x.eap", "ttls");
+        givenMapWith("net.interface.wlan0.config.802-1x.innerAuth", "mschapv2");
+        givenMapWith("net.interface.wlan0.config.802-1x.identity", "example-user-name");
+        givenMapWith("net.interface.wlan0.config.802-1x.password", new Password("secure-test-password-123!@#"));
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild8021xSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenNoExceptionsHaveBeenThrown();
+
+        thenResultingMapContainsArray("eap", new Variant<>(new String[] { "ttls" }).getValue());
+        thenResultingMapContains("phase2-auth", "mschapv2");
+        thenResultingMapContains("identity", "example-user-name");
+        thenResultingMapContains("password", "secure-test-password-123!@#");
+        thenResultingMapNotContains("anonymous-identity");
+        thenResultingMapNotContains("ca-cert");
+        thenResultingMapNotContains("ca-cert-password");
+
+    }
+
+    @Test
+    public void build8021xSettingsTtlsAndMschapV2WithOptionalParams() {
+        givenMapWith("net.interface.wlan0.config.802-1x.eap", "ttls");
+        givenMapWith("net.interface.wlan0.config.802-1x.innerAuth", "mschapv2");
+        givenMapWith("net.interface.wlan0.config.802-1x.anonymous-identity", "anonymous-identity-test-var");
+        givenMapWith("net.interface.wlan0.config.802-1x.ca-cert", "binary ca cert");
+        givenMapWith("net.interface.wlan0.config.802-1x.ca-cert-password", new Password("secure-password"));
+        givenMapWith("net.interface.wlan0.config.802-1x.identity", "example-user-name");
+        givenMapWith("net.interface.wlan0.config.802-1x.password", new Password("secure-test-password-123!@#"));
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild8021xSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenNoExceptionsHaveBeenThrown();
+
+        thenResultingMapContainsArray("eap", new Variant<>(new String[] { "ttls" }).getValue());
+        thenResultingMapContains("phase2-auth", "mschapv2");
+        thenResultingMapContains("anonymous-identity", "anonymous-identity-test-var");
+        thenResultingMapContainsBytes("ca-cert", "binary ca cert");
+        thenResultingMapContains("ca-cert-password", "secure-password");
+        thenResultingMapContains("identity", "example-user-name");
+        thenResultingMapContains("password", "secure-test-password-123!@#");
+
+    }
+
+    @Test
+    public void build8021xSettingsPeapAndMschapV2() {
```

Comment:
Please use the established test naming convention.

```suggestion
    public void build8021xSettingsShouldWorkWithPeapAndMschapV2() {
```